### PR TITLE
Update icu4j to something recent

### PR DESCRIPTION
**Pull request by jvz**
This plugin bundles a positively ancient version of icu4j which has a lot of updates.

### CODE REVIEW

**Comment by jvz** [comment]
I see the only people with access to release this are @stephenc and @batmat.

**Comment by stephenc** [comment]
I approve @jvz to get release permissions for this project

**Comment by jvz** [comment]
@oleg-nenashev do I need to request permission on jenkinsci-dev, or is this comment good enough?

**Comment by oleg-nenashev** [comment]
@jvz this is good enough as long as you reference this comment in https://github.com/jenkins-infra/repository-permissions-updater . After that we will also need to grant you permissions, and it is still better to have a request track in the dev list. But it is not must-have

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1 @@
+buildPlugin()
```
Comment (oleg-nenashev) [inline Jenkinsfile]: You need write permissions to get it running

File: pom.xml [hunk 2]
```diff
@@ -142,21 +145,31 @@
       <artifactId>maven-plugin</artifactId>
       <version>2.5</version>
```
Comment (oleg-nenashev) [inline pom.xml]: Use maven Plugin 3.2, Luke. It fixes some bits of the dependency hell

File: src/main/java/com/cloudbees/plugins/deployer/DeployNowSlave.java [hunk 3]
```diff
@@ -129,7 +129,7 @@ public CauseOfBlockage canTake(Queue.BuildableItem item) {
 
             @Override
             public String getDisplayName() {
-                return null;
+                return "Deploy Now Slave Property";
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/plugins/deployer/DeployNowSlave.java]:
```suggestion
                return "Deploy Now Agent Property";
```

File: src/main/java/com/cloudbees/plugins/deployer/DeployNowSlave.java [hunk 4]
```diff
@@ -111,7 +111,7 @@ public boolean isInstantiable() {
 
         @Override
         public String getDisplayName() {
-            return null;
+            return "Deploy Now Slave";
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/plugins/deployer/DeployNowSlave.java]:
```suggestion
            return "Deploy Now Agent";
```

File: pom.xml [hunk 5]
```diff
@@ -170,41 +183,9 @@
...
-                </rules>
-              </configuration>
-            </execution>
-          </executions>
-        </plugin>
-      </plugins>
-    </pluginManagement>
     <plugins>
       <plugin>
         <artifactId>maven-release-plugin</artifactId>
```
Comment (oleg-nenashev) [inline pom.xml]: Just remove definition from here and rely on parent POM?

File: pom.xml [hunk 6]
```diff
@@ -170,41 +183,9 @@
...
-        <version>2.5.2</version>
       </plugin>
       <plugin>
         <groupId>org.jacoco</groupId>
```
Comment (oleg-nenashev) [inline pom.xml]: remove it as well? Parent POM has a JaCoCo profile

File: Jenkinsfile (hunk 1 above)
Comment (jvz) [reply Jenkinsfile]: That's what I figured, but I thought I could add it for when it was merged to master.

**Comment by jvz** [comment]
@oleg-nenashev all updated

**Pull request merged**